code/code/misc/crit_combat.cc

```diff
@@ -1654,7 +1660,8 @@ buf=format("$n thrusts $s %s deep into $N's torso, impaling $M!") %
 	  desc->career.crit_impale++;
 	if (v->desc)
 	  v->desc->career.crit_impale_suff++;
-	rc = dislodgeWeapon(v, weapon, WEAR_BODY);
+
+	rc = damageLimb(v,WEAR_BODY,weapon,dam);
```

Comment:
What's the story behind removing `dislodgeWeapon`?

Reply:
It's for game balance reasons.  Pierce weapons are objectively worse than all other weapon types in part because you just lose your weapon during combat.  It's a "feature" no one likes as far as I'm aware and from a realism perspective it's just kind of stupid that knife fights would repeatedly lose their knives during a lengthy fight by impaling them inside their opponent.

Reply:
I'm not buying the bit with realism (I've yet to see a computer game of any kind that would be realistic in the slightest).
However, I do empathize with player irritation, so I'm open to killing it.
Seeing as it's a very user-visible change, have you run this by the design department yet?
Is anything else calling dislodgeWeapon now, or should we delete that code?

Reply:
I think the gameplay concerns are reason enough to remove the dislodge effect from pierce crits; the comment on realism (or lack thereof) for constantly losing pierce weapons in a fight was simply the icing on the cake.

I've run these changes by others before submitting but I'd welcome feedback here if people have it, and I've asked for a few to chime in here with their feedback from a gameplay perspective.

As far as whether other functions are using it, I'll have to take a look and get back to you on that.

Reply:
@cizra we've had several discussions on the removal of dislodge weapon.  The design department is for it and sees this as a minor tweak to address a major irritation.